### PR TITLE
EWPP-1257: Language support in the Persistent URL controller and fixes on PURL redirection.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

**Review by imanoleguskiza — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

File: modules/oe_content_persistent/src/Controller/PersistentUrlController.php [hunk 1]
```diff
@@ -65,16 +72,17 @@ public static function create(ContainerInterface $container) {
...
+    $langcode = $this->languageManager->getCurrentLanguage()->getId();
+    if ($entity = $this->contentUuidResolver->getEntityByUuid($uuid, $langcode)) {
       if ($entity instanceof ContentEntityInterface) {
         $url = $this->contentUrlResolver->resolveUrl($entity);
-        return new RedirectResponse($url->toString(), 302, ['PURL' => '1']);
+        $generated_url = $url->toString(TRUE);
+        $cache = CacheableMetadata::createFromObject($generated_url);
+        $cache->addCacheableDependency($entity);
+        $cache->addCacheContexts(['url', 'languages']);
+        $response = new TrustedRedirectResponse($generated_url->getGeneratedUrl(), 302, ['PURL' => '1']);
```
Comment (sergepavle) [inline modules/oe_content_persistent/src/Controller/PersistentUrlController.php]: Looks like we don't need `'PURL' => '1'` anymore.

Comment (sergepavle) [inline modules/oe_content_persistent/src/Controller/PersistentUrlController.php]:
```suggestion
        $response = new TrustedRedirectResponse($generated_url->getGeneratedUrl());
```

File: modules/oe_content_persistent/src/Controller/PersistentUrlController.php [hunk 2]
```diff
@@ -65,16 +72,17 @@ public static function create(ContainerInterface $container) {
...
-      // https://www.lullabot.com/articles/early-rendering-a-lesson-in-debugging-drupal-8
+    $langcode = $this->languageManager->getCurrentLanguage()->getId();
+    if ($entity = $this->contentUuidResolver->getEntityByUuid($uuid, $langcode)) {
       if ($entity instanceof ContentEntityInterface) {
         $url = $this->contentUrlResolver->resolveUrl($entity);
-        return new RedirectResponse($url->toString(), 302, ['PURL' => '1']);
+        $generated_url = $url->toString(TRUE);
+        $cache = CacheableMetadata::createFromObject($generated_url);
+        $cache->addCacheableDependency($entity);
+        $cache->addCacheContexts(['url', 'languages']);
```
Comment (sergepavle) [inline modules/oe_content_persistent/src/Controller/PersistentUrlController.php]: Probably better to use a more specific cache context like `url.path`.

**Review by sergepavle — CHANGES_REQUESTED** [review]

**Comment by sergepavle** [comment]
As far as the core issue mentioned here https://www.lullabot.com/articles/early-rendering-a-lesson-in-debugging-drupal-8 looks like fixed, we can remove modules/oe_content_persistent/src/EventSubscriber/PersistentUrlRedirectSubscriber.php event subscriber.

File: modules/oe_content_persistent/src/Controller/PersistentUrlController.php (hunk 2 above)
Comment (upchuk) [reply modules/oe_content_persistent/src/Controller/PersistentUrlController.php]: It's ok, let's keep it as wide as possible.

**Review by sergepavle — DISMISSED** [review]

**Review by sergepavle — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Pull request merged**